adafruit_rgb_display/rgb.py

```diff
@@ -71,74 +82,90 @@ def _decode_pixel(self, data):
         """Decode bytes into a pixel color."""
         return color565(*struct.unpack(self._DECODE_PIXEL, data))
 
-    def pixel(self, x, y, color=None):
+    def pixel(self, x, y, color=None): #pylint: disable-msg=invalid-name
         """Read or write a pixel."""
         if color is None:
             return self._decode_pixel(self._block(x, y, x, y))
-        if not 0 <= x < self.width or not 0 <= y < self.height:
-            return
-        self._block(x, y, x, y, self._encode_pixel(color))
 
-    def fill_rectangle(self, x, y, width, height, color):
+        if 0 <= x < self.width and 0 <= y < self.height:
+            self._block(x, y, x, y, self._encode_pixel(color))
+        return None
+
+    #pylint: disable-msg=too-many-arguments
+    def fill_rectangle(self, x, y, width, height, color):  #pylint: disable-msg=invalid-name
         """Draw a filled rectangle."""
         x = min(self.width - 1, max(0, x))
         y = min(self.height - 1, max(0, y))
-        w = min(self.width - x, max(1, width))
-        h = min(self.height - y, max(1, height))
-        self._block(x, y, x + w - 1, y + h - 1, b'')
-        chunks, rest = divmod(w * h, _BUFFER_SIZE)
+        width = min(self.width - x, max(1, width))
+        height = min(self.height - y, max(1, height))
+        self._block(x, y, x + width - 1, y + height - 1, b'')
+        chunks, rest = divmod(width * height, _BUFFER_SIZE)
         pixel = self._encode_pixel(color)
         if chunks:
             data = pixel * _BUFFER_SIZE
-            for count in range(chunks):
-                self._write(None, data)
-        self._write(None, pixel * rest)
+            for _ in range(chunks):
+                self.write(None, data)
+        self.write(None, pixel * rest)
+    #pylint: enable-msg=too-many-arguments
 
     def fill(self, color=0):
         """Fill whole screen."""
         self.fill_rectangle(0, 0, self.width, self.height, color)
 
-    def hline(self, x, y, width, color):
+    def hline(self, x, y, width, color): #pylint: disable-msg=invalid-name
         """Draw a horizontal line."""
         self.fill_rectangle(x, y, width, 1, color)
 
-    def vline(self, x, y, height, color):
+    def vline(self, x, y, height, color): #pylint: disable-msg=invalid-name
         """Draw a vertical line."""
         self.fill_rectangle(x, y, 1, height, color)
 
+    def write(self, command, data):
+        """Derived class must implement this"""
+        raise NotImplementedError
+
+    def read(self, command, count):
+        """Derived class must implement this"""
+        raise NotImplementedError
```

Comment:
I would advise against dummy methods raising `NotImplementedError`. Not only they waster memory, but they also make it very hard to use `super()` properly. I think that `AttributeError` in case something tries to call this is good enough.

Reply:
I believe pylint has smarts around recognizing unimplemented methods like this and validating that suubclasses define them. The memory hit won't be so large that it isn't worth it.

Reply:
The real problem is when you use this class in any multiple inheritance scenario, where you *have* to use `super()` to make sure all the methods are properly called, but this will also call this method and raise a `NotImplementedError`. That's why this is an anti-pattern. I understand that with CircuitPython we run a much lower risk for this, but it's still a bad habit.

Reply:
Oh, interesting! What would you recommend in this case? Would you mind opening an issue for us to make the drivers consistent on how to do this?

Reply:
The way to do this in normal Python is to use `@abc.abstractmethod`, but CircuitPython doesn't have the `abc` module. Personally I'm inclined to just leaving this out completely, and just mentioning in the comments that the subclasses need to implement those methods. This is not something the user of the library should ever see or need anyways.